internal/compiler/parser.rs

```diff
@@ -12,7 +12,11 @@ This module has different sub modules with the actual parser functions
 
 */
 
-use crate::diagnostics::{BuildDiagnostics, SourceFile, Spanned};
+use crate::{
+    diagnostics::{BuildDiagnostics, SourceFile, Spanned},
+    expression_tree::Expression,
+    lookup::LookupCtx,
```

Comment:
Would be nice if the parser did not depend on the lower-level datastructure.

Ideally the parser would just create a syntax node. 

Then further analysis should be done in the LSP itself, i'd say.  (and Expression::from_binding_expression_node could be pub)